dds/DCPS/DataWriterImpl.cpp

```diff
@@ -1345,7 +1345,7 @@ DataWriterImpl::enable()
                                            qos_.reliability.max_blocking_time,
                                            n_chunks_,
                                            domain_id_,
-                                           get_topic_name(),
+                                           topic_name_.in(),
```

Comment:
topic_name_ is a CORBA::String_var, the constructor of WriteDataContainer expects a const char*, so we have should use .in(). Could be that it compiles without .in() with gcc but we had problems with that with TAO with some compilers so I always use .in().